packages/hmr/README.md

```diff
@@ -0,0 +1,179 @@
+[![GitHub](https://img.shields.io/github/license/sapphiredev/plugins)](https://github.com/sapphiredev/plugins/blob/main/LICENSE.md)
+[![codecov](https://codecov.io/gh/sapphiredev/plugins/branch/main/graph/badge.svg?token=QWL8FB16BR)](https://codecov.io/gh/sapphiredev/plugins)
+[![npm bundle size](https://img.shields.io/bundlephobia/min/@sapphire/plugin-hmr?logo=webpack&style=flat-square)](https://bundlephobia.com/result?p=@sapphire/plugin-hmr)
+[![npm](https://img.shields.io/npm/v/@sapphire/plugin-hmr?color=crimson&logo=npm&style=flat-square)](https://www.npmjs.com/package/@sapphire/plugin-hmr)
+
+</div>
+
+## Description
+
+This plugin provides a HMR (Hot Module Replacement) feature for @sapphire/framework. This allows you to add, delete, and
+update commands, listeners, and more without having to restart your bot. This allows for rapid iteration and prototyping
+without your bot slowing you down.
+
+## Features
+
+-   Fully ready for TypeScript!
+-   Includes ESM ready entrypoint
+-   FAST!
+-   Super easy to use
+
+## Installation
+
+`@sapphire/plugin-hmr` depends on the following packages. Be sure to install these along with this package!
+
+-   [`@sapphire/framework`](https://www.npmjs.com/package/@sapphire/framework)
+
+You can use the following command to install this package, or replace `npm install` with your package manager of choice.
+
+```sh
+npm install @sapphire/plugin-hmr @sapphire/framework
+```
+
+---
+
+## Usage
+
+### JavaScript
+
+in your main or setup file, register the plugin:
+
+```javascript
+require('@sapphire/plugin-hmr/register');
+```
+
+if you want to make sure its only loaded in development, you can do this:
+
+```javascript
+const { start } = require('@sapphire/hmr');
+
+const client = new SapphireClient({
+	/* your bot options */
+});
+
+async function main() {
+	await client.login();
+
+	// this has to be called after you have logged into your bot
+	if (process.env.NODE_ENV === 'development') start();
+}
+
+main();
+```
+
+### TypeScript
+
+in your main or setup file, register the plugin:
+
+```typescript
+import '@sapphire/plugin-hmr/register';
+```
+
+if you want to make sure its only loaded in development, you can do this:
+
+```typescript
+import { start } from '@sapphire/plugin-hmr';
+
+const client = new SapphireClient({
+	/* your bot options */
+});
+
+async function main() {
+	await client.login();
+
+	// this has to be called after you have logged into your bot
+	if (process.env.NODE_ENV === 'development') start();
+}
+
+void main();
+```
+
+In order for proper HMR, you will need to recompile your code, so you will need to edit your package.json:
+
+- replace the dev and start script 
+```json
+"scripts": {
+	"dev": "run-p watch:build start",
+	"start": "run-s build start:bot"
+}
+```
+
+- add a start:bot script
+```json
+"scripts": {
+	"start:bot": "node dist/index.js"
+}
+```
+
+then run your bot using the dev script. You can replace npm with your package manager of choice.
+```sh
+npm run start
+```
+
```

Comment:
```suggestion
_Note:_ This uses the `run-p` script which is part of [`npm-run-all`](https://github.com/mysticatea/npm-run-all)
_Note 2:_ Please do note that because the processes are started simultaneously you should run `build` at least once before running `dev`, otherwise the `start` process will fail because there are no files to run just yet.
```